src/components/ListItem.tsx

```diff
@@ -1,8 +1,70 @@
-import * as api from '../api';
-import './ListItem.css';
+import "./ListItem.css";
+import { updateItem, ListItem } from "../api";
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { is24HoursLater } from "../utils";
 
-type Props = Pick<api.ListItem, 'name'>;
+interface Props {
+	item: ListItem;
+	listPath: string | null;
+}
+
+export function ListItemCheckBox({ item, listPath }: Props) {
+	const [checked, setChecked] = useState(false);
+
+	const purchaseDate = item.dateLastPurchased
+		? item.dateLastPurchased.toDate()
+		: item.dateLastPurchased;
+
+	useEffect(() => {
+		if (purchaseDate) {
+			setChecked(!is24HoursLater(purchaseDate));
+		} else {
+			setChecked(false);
+		}
+	}, [item.dateLastPurchased]);
+
+	const handleCheckChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+		const isChecked = e.target.checked;
+
+		if (!isChecked && purchaseDate && !is24HoursLater(purchaseDate)) {
```

Comment:
There's a bit of a "code smell" here that I think is worth thinking about. The idea "state modeling" and "the flow of data" are thoughts I tend to obsess a little about when designing React components. This is an excellent example of why I feel so strongly about the value of that. 

Take a close look at at this condition: how quickly can you figure out what its value is when reading this component, especially as you think about how the inputs could be modifying the value of `isChecked`? I'd argue it takes a bit too long to get to the bottom of that, which is the code smell. `isChecked` could be false because of the `setChecked` on line 23, but it also shares a very close relationship with `purchaseDate`, which here is _also_ being checked.. but does it to be? Wouldn't `isChecked` already be true if `purchaseDate && !is24HoursLater(purchaseDate)` evaluates to `true` because of the check on line 20? I believe this is the same as saying `if(!isChecked && !isChecked)`

Deriving state of props, like we're doing with `isChecked`, can be helpful, but if at any point that state you derived wasn't enough and you start deriving other state on the fly, things can get both very difficult to understand and potentially incorrect. I've got lots more thoughts here, but it's getting late 😴 we can talk sometime soon about all that if you'd like! For now, I think 2/3 of this condition is redundant.

Reply:
I'd love to talk about this more and just get more of your insight!